app/lib/bk/compat/parsers/circleci.rb

```diff
@@ -45,16 +45,20 @@ def self.matches?(text)
           # legacy format support
           config.fetch('version', 2.1) == 2 && config['jobs'].include?('build')
         end
+      rescue Psych::SyntaxError
+        false
       end
 
       def initialize(text, options = {})
-        @config = YAML.safe_load(text, aliases: true)
-        @now = Time.now
-        @options = options
-        @commands_by_key = {}
-        @executors = {}
-
-        register_translators!
+        BK::Compat::Error::CompatError.safe_yaml do
+          @config = YAML.safe_load(text, aliases: true)
```

Comment:
only this line should be part of the block and (as mentioned in other places) if you create a helper method, it would be a lot less change:

```suggestion
          @config = MODULE::load_yaml(text, aliases: true)
```